src/react/databrowser/buckets/__tests__/BucketList.test.jsx

```diff
@@ -0,0 +1,82 @@
+import BucketList from '../BucketList';
+import { MemoryRouter } from 'react-router-dom';
+import React from 'react';
+import { reduxMount } from '../../../utils/test';
+
+describe('BucketList', () => {
+    const buckets = [{
+        CreationDate: 'Wed Oct 07 2020 16:35:57',
+        LocationConstraint: 'us-east-1',
+        Name: 'bucket1',
+    }, {
+        CreationDate: 'Wed Oct 07 2020 16:35:57',
+        LocationConstraint: 'us-east-1',
+        Name: 'bucket2',
+    }];
+
+    it('should list buckets with the data associated with', () => {
+        const { component } = reduxMount(
+            <MemoryRouter>
+                <BucketList buckets={buckets} locations={{
+                    'us-east-1': {
+                        isBuiltin: true,
+                        locationType: 'location-file-v1',
+                        name: 'us-east-1',
+                        objectId: '',
+                    },
+                }} selectedBucketName="bucket1"/>
+            </MemoryRouter>,
+        );
+
+        const firstBucketCellLink = component.find('Cell').at(0).find('a');
+        const firstBucketCellLocation = component.find('Cell').at(1);
+        const firstBucketCellDate = component.find('Cell').at(2);
+        expect(firstBucketCellLink.text()).toBe('bucket1');
+        expect(firstBucketCellLink.prop('href')).toBe('/buckets/bucket1/objects');
+        expect(firstBucketCellLocation.text()).toBe('us-east-1 / Zenko Local Filesystem');
+        expect(firstBucketCellDate.text()).toBe('Wed Oct 07 2020 16:35:57');
+
+        const secondBucketCellLink = component.find('Cell').at(3).find('a');
+        const secondBucketCellLocation = component.find('Cell').at(4);
+        const secondBucketCellDate = component.find('Cell').at(5);
+        expect(secondBucketCellLink.text()).toBe('bucket2');
+        expect(secondBucketCellLink.prop('href')).toBe('/buckets/bucket2/objects');
+        expect(secondBucketCellLocation.text()).toBe('us-east-1 / Zenko Local Filesystem');
+        expect(secondBucketCellDate.text()).toBe('Wed Oct 07 2020 16:35:57');
+    });
+
+    it('should select row if the bucket name specified in the parameter matches one of the bucket names listed', () => {
+        const { component } = reduxMount(
+            <MemoryRouter>
+                <BucketList buckets={buckets} locations="" selectedBucketName="bucket2"/>
+            </MemoryRouter>,
+        );
+
+        const bucketRows = component.find('Row');
+        bucketRows.forEach((bucketRow, index) => {
+            if (buckets[index].Name === 'bucket1') {
```

Comment:
We don't know that `bucketRows` and `buckets` arrays will stay sorted in the same way. 
If we decide to sort our `<Row>`s by bucket name for example, code will still be correct (meaning that the right `<Row>` will be selected) but the tests will fail.

Reply:
done